libraries/APM_Control/AR_AttitudeControl.cpp

```diff
@@ -432,8 +432,8 @@ float AR_AttitudeControl::get_throttle_out_stop(bool motor_limit_low, bool motor
         // could not get speed so assume stopped
         _stopped = true;
     } else {
-        // if vehicle drops below _stop_speed consider it stopped
-        if (fabsf(speed) <= fabsf(_stop_speed)) {
+        // if desired speed is zero and vehicle drops below _stop_speed consider it stopped
+        if (is_zero(_desired_speed) && fabsf(speed) <= fabsf(_stop_speed)) {
```

Comment:
should this also be fabsf(_desired_speed) <= fabsf(_stop_speed) ?

Reply:
It's a good question but I think I'll leave it at requiring the desired_speed to be exactly zero.  I think when the caller wants the vehicle to stop they will request a desired_speed of exactly zero.